datasetviewer/stack/StackPresenter.py

```diff
@@ -0,0 +1,388 @@
+            Defaults to None as it set during `_set_dict`.
+
+    Raises:
+        ValueError: If the StackView or the DimensionViewFactory are None.
+    """
+
+    def __init__(self, stack_view, dim_fact):
+
+        if stack_view is None:
+            raise ValueError("Error: Cannot create StackView when View is None.")
+
+        if dim_fact is None:
+            raise ValueError("Error: Cannot create DimensionViewFactory when View is None.")
+
+        self._view = stack_view
+        self._dim_fact = dim_fact
+        self._dict = None
+        self._master = None
+        self._dim_presenters = {}
+        self._current_stack_face = None
+        self._stack_idx = {}
+
+    def register_master(self, master):
+        """
+
+        Register the MainViewPresenter as the StackPresenter's master, and subscribe the MainViewPresenter to the
+        StackPresenter.
+
+        Args:
+            master (MainViewPresenter): An instance of a MainViewPresenter.
+
+        """
+        assert (isinstance(master, MainViewPresenterInterface))
+
+        self._master = master
+        master.subscribe_stack_presenter(self)
+
+    def set_dict(self, dict):
+        """
+
+        Assigns the _dict variable in the StackPresenter, creates the required amount of Dimension widgets, and then
+        places these widgets on the StackView.
+
+        Args:
+            dict (DataSet): An OrderedDict of xarray Datasets.
+
+        """
+
+        self._dict = dict
+
+        # Clear the DimensionViews that were on the Stack
+        self._clear_stack()
+
+        # Clear the information about the previous presenters/widgets
+        self._dim_presenters = {}
+        self._stack_idx = {}
+
+        for key in dict.keys():
+
+            # Create a new Stack element for every dataset in the dictionary
+            idx = self._view.create_stack_element()
+
+            # Record which index on the Stack this dataset has
+            self._stack_idx[key] = idx
+
+            # Prepare a dictionary for the presenters for this dataset will use
+            self._dim_presenters[key] = {}
+
+            data = dict[key].data
+
+            # Ignore data with a single dimension as they will not need Dimension widgets
+            if len(data.dims) > 1:
+
+                for i in range(len(data.dims)):
+
+                    # Create a set of widgets for every dimension in the dataset
+                    w = self._dim_fact.create_widgets(data.dims[i], data.shape[i])
+
+                    # Obtain the presenter for the widget
+                    self._dim_presenters[key][data.dims[i]] = w.get_presenter()
+
+                    # Store the presenter in the dictionary using the key and the dimension name
+                    self._dim_presenters[key][data.dims[i]].register_stack_master(self)
+
+                    # Place the widgets in the StackView's GridLayout
+                    for j, widget in enumerate(w.get_widgets()):
```

Comment:
I found that using `DimensionWidget` as a widget with a horizontal layout made the sliders and buttons misaligned between the different dimensions. Placing the individual widgets into a `GridLayout` was the only way I found to avoid this.

Reply:
This is not the right solution. You could try setting a fixed width for each item in the layout of the individual `DimensionWidget`.